tests/driver_lsm6dsl/main.c

```diff
@@ -12,28 +12,590 @@
+         == 0) {
+        return _lsm6dsl_cfg_wkup_int_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "ff_int") == 0) {
+        return _lsm6dsl_cfg_ff_int_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "6d4d_int") == 0) {
+        return _lsm6dsl_cfg_6d4d_int_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "tap_int") == 0) {
+        return _lsm6dsl_cfg_tap_int_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "ofs_xyz") == 0) {
+        return _lsm6dsl_cfg_ofs_xyz_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "ofs_x") == 0) {
+        return _lsm6dsl_cfg_ofs_x_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "ofs_y") == 0) {
+        return _lsm6dsl_cfg_ofs_y_handler(argc -1, &argv[1]);
+    }
+    else if (strcmp(argv[1], "ofs_z") == 0) {
+        return _lsm6dsl_cfg_ofs_z_handler(argc -1, &argv[1]);
+    }
+    else {
+        _usage_configure();
+    }
+    return 0;
+}
+
+static void _usage_enable(void)
+{
+    printf("Usage: enable <wkup_int | sleep_int | 6d4d_int | ff_int | " \
+            "tap1_int | tap2_int> <pin> \n");
+    puts("");
+}
+
+static int _lsm6dsl_enable_handler(int argc, char **argv)
+{
+    if (argc < 3) {
+        _usage_enable();
+        return 1;
+    }
+    if (strcmp(argv[1], "wkup_int") == 0) {
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_WAKE_UP, atoi(argv[2]),
+            alert_cb, "WakeUp") != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+        puts(" To trig the alert, move the device ...");
+    }
+    else if (strcmp(argv[1], "sleep_int") == 0) {
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_SLEEP, atoi(argv[2]),
+            alert_cb, "Sleep") != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+        puts(" To trig the alert, move the device and stop, wait configured " \
+                "sleep duration");
+    }
+    else if (strcmp(argv[1], "ff_int") == 0) {
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_FREE_FALL,
+            atoi(argv[2]), alert_cb, "FreeFall") != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+        puts(" To trig the alert, release your device at the right height " \
+                "to ger a free-fall...");
+    }
+    else if (strcmp(argv[1], "6d4d_int") == 0) {
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_6D_4D, atoi(argv[2]),
+            alert_cb, "6D4D") != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+        puts(" To trig the alert, change the device orientation/inclination");
+    }
+    else if (strcmp(argv[1], "tap1_int") == 0) {
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_SINGLE_TAP,
+            atoi(argv[2]), alert_cb, "SingleTAP") != LSM6DSL_OK) {
```

Comment:
The `tap1_int` and `tap2_int` sub-commands expect a parameter but it's not documented in the usage function.